File: web-basics/WebLab3/src/main/java/com/arslanka/backend/models/views/Circle.java
package com.arslanka.backend.models.views;

public class Circle implements Shape {
    private final double multiplier;

    public Circle(Distance radius) {
        if (radius == Distance.HALF_R) {
            multiplier = 1.0d / 2.0d;
        } else {
            multiplier = 1.0d;
        }
    }

    @Override
    public Boolean isHit(Double radius, Double x, Double y) {
        return (x * x + y * y) <= ((radius * multiplier) * (radius * multiplier));
    }
}

File: web-basics/WebLab4/src/main/kotlin/com/example/arslanka/weblab4/models/views/Circle.kt
package com.example.arslanka.weblab4.models.views

import java.math.BigDecimal

class Circle(radius: Distance) : Shape {
    private var multiplier: BigDecimal

    init {
        multiplier = if (radius == Distance.HALF_R)
            BigDecimal.valueOf(0.5)
        else
            BigDecimal.ONE
    }

    override fun isHit(radius: BigDecimal, x: BigDecimal, y: BigDecimal): Boolean {
        return (x * x + y * y) <= ((radius * multiplier) * (radius * multiplier))
    }
}
